### PR TITLE
Dance page

### PR DESCRIPTION
**Pull request by H-R-G-P**

### CODE REVIEW

File: src/Controller/DanceController.php [hunk 1]
```diff
@@ -50,37 +53,45 @@ public function index(DanceRepository $danceRepository): Response
...
             return new Response('This dance dose not exists.');
         }
 
+        $versions = $versionRepository->findBy([
+            'dance' => $dance->getId(),
+        ]);
+        // TODO Move sorting to VersionRepository method
+        usort($versions, function ($a, $b) {
+            return strcmp($a->getName(), $b->getName());
+        });
```
Comment (b1514) [inline src/Controller/DanceController.php:76]: Move sorting to 'findBy' method

File: src/Controller/DanceController.php [hunk 2]
```diff
@@ -50,37 +53,45 @@ public function index(DanceRepository $danceRepository): Response
      * @param DanceRepository<Dance> $danceRepository
      * @param VersionRepository<Version> $versionRepository
      * @param EntityManagerInterface $entityManager
+     * @param MapService $mapService
+     * @param PlaceRepository<Place::class> $placeRepository
      *
      * @return Response
      */
-    public function show(string $slug, RequestStack $requestStack, DanceRepository $danceRepository, VersionRepository $versionRepository, EntityManagerInterface $entityManager): Response
+    public function show(string $slug, RequestStack $requestStack, DanceRepository $danceRepository, VersionRepository $versionRepository, EntityManagerInterface $entityManager, MapService $mapService, PlaceRepository $placeRepository): Response
```
Comment (b1514) [inline src/Controller/DanceController.php:61]: To many lines

File: src/Service/MapService.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+     *
+     * @return ?MapDTO
+     */
+    public function createMapDTO(array $places): ?MapDTO
+    {
+        $map = null;
+        $points = [];
+        foreach ($places as $place) {
+            if ($place->getLat() !== null && $place->getLon() !== null){
+                $coordinates = new CoordinatesVO($place->getLat(), $place->getLon());
```
Comment (b1514) [inline src/Service/MapService.php:27]: Place->getCoordinates() return  ?CoordinatesVO

File: src/Vo/AddressVO.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    /** As possible full address to village.
+     * If village unknown, then only department and region, itp.
+     *
+     * @var string
+     *
+     * @example "Мінская вобл., Барысаўскі р-н, в. Селішча"
+     */
+    private string $address;
+
+    public function __construct(Version $v)
```
Comment (b1514) [inline src/Vo/AddressVO.php:20]: Add language parameter

File: src/Vo/CoordinatesVO.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
+<?php declare(strict_types=1);
+
+
+namespace App\Vo;
+
+
+class CoordinatesVO
+{
+    /** Latitude
+     * @var float
```
Comment (b1514) [inline src/Vo/CoordinatesVO.php:10]: Do not var line, if parameter have it 'private float...'

File: tests/Service/MapServiceTest.php [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+        try {
+            $polygon = null;
+            $polygon = $service->createPolygonVO($points);
+            self::assertSame(null, $polygon);
+        }catch (Exception $e) {
+            self::assertSame('Minimum 2 places.', $e->getMessage());
+        }
+
+        $points[] = new CoordinatesVO(1.4, 1.6);
```
Comment (b1514) [inline tests/Service/MapServiceTest.php:37]: Add index for readable of test

File: src/Controller/DanceController.php (hunk 2 above)
Comment (H-R-G-P) [reply src/Controller/DanceController.php]: Make 2 less

File: src/Service/MapService.php (hunk 3 above)
Comment (H-R-G-P) [reply src/Service/MapService.php:27]: Do you want add this method to Place entity?

Comment (b1514) [reply src/Service/MapService.php:27]: Yes

File: templates/dance/show_version.html.twig [hunk 7]
```diff
@@ -52,7 +52,7 @@
             </div>
         </div>
     {% endif %}
-
+
```
Comment (b1514) [inline templates/dance/show_version.html.twig:55]: Empty string?

Comment (b1514) [inline templates/dance/show_version.html.twig:55]: Empty string?

**Review by b1514 — APPROVED** [review]

**Pull request closed without merging**